### PR TITLE
[Dem]review benchmarks

### PR DESCRIPTION
**Pull request by farrufat-cimne**

### CODE REVIEW

**Comment by farrufat-cimne** [comment]
added fix for cluster destructor

**Comment by maceligueta** [comment]
@farrufat-cimne , I see 209 files changed for a small code modification. I think it's due to a folder renaming.
Let's try this
`git mv test_examples benchmarks`
instead of changing the name outside git. In this way maybe the PR will be cleaner and, if it works, we'll learn something new.

**Comment by maceligueta** [comment]
Forget my last comment, I tried myself and apparently it changes the name of each file, exactly the same as what we had.

**Comment by farrufat-cimne** [comment]
fixed codacy errors

**Comment by GuillermoCasas** [comment]
@farrufat-cimne Please resolve the conflicts.

**Comment by maceligueta** [comment]
There's a file called '.directory' that should not be there, I think it comes from the Operating System.

File: applications/DEMApplication/benchmarks/multiprocess_benchmarking_full.py [hunk 1]
```diff
@@ -81,18 +84,20 @@ def worker(queue):
...
+            #     g.write("\n===== THORNTON PAPER TESTS. FULL REGIME. LINEAR LAW =====\n\n")
+            # if benchmark == 11:
+            #     g.write("\n===== THORNTON PAPER TESTS. FULL REGIME. HERTZIAN LAW ===\n\n")
+            # if benchmark == 12:
+            #     g.write("\n===== WENSRICH PAPER TEST. ROLLING FRICTION =============\n\n")
+            # if benchmark == 13:
+            #     g.write("\n===== DE/FE CONTACT BENCHMARKS ==========================\n\n")
+            # if benchmark == 20:
+            #     g.write("\n===== BASIC CONTINUUM TESTS  ============================\n\n")
+            # if benchmark == 30:
```
Comment (GuillermoCasas) [inline applications/DEMApplication/benchmarks/multiprocess_benchmarking_full.py]: Please, either remove these lines or not, do not just comment them.

File: applications/DEMApplication/benchmarks/multiprocess_benchmarking_full.py [hunk 2]
```diff
@@ -101,25 +106,28 @@ def main():
...
+        #g.write("\n=========== DEM DISCONTINUUM TESTS ===========\n")
+        #g.write("\n==== TSUJI PAPER BENCHMARKS. SLIDING REGIME ==\n\n")
         g.close()
         Text = ""
         failure = False
 
         q = queue.Queue()
 
-        #Discontinuum Tests. From 1 to 17
+        #### Discontinuum Tests. From 1 to 17
```
Comment (GuillermoCasas) [inline applications/DEMApplication/benchmarks/multiprocess_benchmarking_full.py]: unnecessary

File: applications/DEMApplication/python_scripts/DEM_benchmarks_class.py [hunk 3]
```diff
@@ -4203,11 +4213,828 @@ def compute_errors(self, velocity_list_outfile_name):  #FINALIZATION STEP
...
+            for node in modelpart.Nodes:
+                if node.Id == 42:           ### stage 1
+
+                    force_node_x = node.GetSolutionStepValue(TOTAL_FORCES)[0]
+                    force_node_y = node.GetSolutionStepValue(TOTAL_FORCES)[1]
+                    force_node_z = node.GetSolutionStepValue(TOTAL_FORCES)[2]
+
+                    angular_node_x = node.GetSolutionStepValue(ANGULAR_VELOCITY)[0]
+                    angular_node_y = node.GetSolutionStepValue(ANGULAR_VELOCITY)[1]
+                    angular_node_z = node.GetSolutionStepValue(ANGULAR_VELOCITY)[2]
```
Comment (GuillermoCasas) [inline applications/DEMApplication/python_scripts/DEM_benchmarks_class.py:4284]: Could you make this type of assignments more concise? There are many of them in this file

File: applications/DEMApplication/python_scripts/DEM_benchmarks_class.py [hunk 4]
```diff
@@ -4203,11 +4213,828 @@ def compute_errors(self, velocity_list_outfile_name):  #FINALIZATION STEP
...
+
+                    displacement = [0]*3
+                    displacement[0] = node.X - node.X0
+                    displacement[1] = node.Y - node.Y0
+                    displacement[2] = node.Z - node.Z0
+
+                    force_node = sqrt(force_node_x*force_node_x + force_node_y*force_node_y + force_node_z*force_node_z)
```
Comment (GuillermoCasas) [inline applications/DEMApplication/python_scripts/DEM_benchmarks_class.py:4367]: Could you not just implement a modulus function?

File: applications/DEMApplication/python_scripts/DEM_benchmarks_class.py [hunk 5]
```diff
@@ -4203,11 +4213,828 @@ def compute_errors(self, velocity_list_outfile_name):  #FINALIZATION STEP
...
+                    angular_node = sqrt(angular_node_x*angular_node_x + angular_node_y*angular_node_y + angular_node_z*angular_node_z)
+                    displacement_node = sqrt(displacement[0]*displacement[0] + displacement[1]*displacement[1] + displacement[2]*displacement[2])
+
+                    i=8
+                    data  = open("benchmark" + str(sys.argv[1]) + "_graph%s.dat" % i, 'a')
+                    data.write(str("%.8g"%time).rjust(12)+" "+str("%.6g"%force_node).rjust(13)+" "+str("%.6g"%angular_node).rjust(13)+" "+str("%.6g"%displacement_node).rjust(13)+"\n")
```
Comment (GuillermoCasas) [inline applications/DEMApplication/python_scripts/DEM_benchmarks_class.py:4475]: Could you encapsulate tyhis kind of writes? They seem to repeat often

File: applications/DEMApplication/test_examples/basic_benchmarks/paper_data/.directory [hunk 6]
```diff
@@ -1,5 +0,0 @@
-[Dolphin]
```
Comment (GuillermoCasas) [inline applications/DEMApplication/test_examples/basic_benchmarks/paper_data/.directory:1]: Please remove this directory as discussed

**Review by GuillermoCasas — CHANGES_REQUESTED** [review]

**Pull request closed without merging**